src/main/java/frc/robot/Robot.java

```diff
@@ -142,6 +142,58 @@ public void robotInit() {
                 }, Set.of(armTelescopicSystem, clawJointSystem, clawGripperSystem)
         );
 
+
+        Command collectFromSourceCommand = new ParallelCommandGroup(
+                new ArmTelescopicMoveToLength(armTelescopicSystem,
+                       RobotMap.CALCULATION_COLLECT_FROM_SOURCE),
+                Commands.runOnce(()->  armJointControlCommand.setTargetPosition(Math.atan(RobotMap.SOURCE_HEIGHT-RobotMap.SWERVE_HEIGHT/RobotMap.SWERVE_DISTANCE_FROM_CORAL))),
+                Commands.waitUntil(()-> armJointControlCommand.isAtTargetPosition()),
+                new MoveClawJointToPosition(clawJointSystem,RobotMap.CLAWJOINT_SOURCE_ANGLE)
+        );
+
+        Command placeCoralOnReefCommandL1 = new ParallelCommandGroup(
```

Comment:
instead of having 4 copy-paste commands for the reef with little changes, create a method 
`private Command placeCoralOnReefCommandSimple(CoralReef coralReef)`

In the command, use a switch case on `coralReef` to get all the variables depending on which reef it is, and then create the command group and return it. This will just remove all the duplication of the 4 different command groups.

See `placeCoralOnReefCommand` as an example